pypuf/simulation/arbiter_based/ltfarray.py

```diff
@@ -223,6 +223,39 @@ def transform_polynomial(cs, k):
         assert result.shape == (N, k, n), 'The resulting challenges have not the desired shape. Sorry!'
         return result
 
+    @staticmethod
+    def transform_permutation_atf(cs, k):
+        """
+        This transformation performs first a pseudorandom permutation of the challenge k times before applying the
+        ATF transformation to each challenge.
+        :param cs:
+        :param k:
+        :return:
+        """
+        N = len(cs)
+        n = len(cs[0])
+        seed = 0x1234
+
+        """ Perform random permutations """
+        cs_permuted = array(
+            [
+                [RandomState(seed + i).permutation(c)
+                 for i in range(k)]
+                for c in cs
+                ]
+        )
+        """ Perform atf transform """
+        result = transpose(
+            array([
+                      prod(cs_permuted[:, :, i:], 2)
+                      for i in range(n)
+                      ]),
+            (1, 2, 0)
```

Comment:
`transform_atf` takes as input an array (usually with the shape (N, n)) and blows it up to (N, k, n). In our case we need to apply the `transform_atf` to an array with shape (N, k, n) and not change the shape of the array. Therefore the second parameter (1, 2, 0) for the `np.transponse` function.

To use `transform_atf` it would be necessary to rewrite the function to check for the shape of the `cs` parameter and to apply the corresponding second argument to `np.transponse`.